dotnet/samples/Concepts/FunctionCalling/AzureOpenAI_FunctionCallingTelemetry.cs

```diff
@@ -0,0 +1,274 @@
+        IKernelBuilder kernelBuilder = Kernel.CreateBuilder();
+        kernelBuilder.Services.AddSingleton(loggerFactory);
+        kernelBuilder.Services.AddSingleton<IFunctionInvocationFilter>(functionFilter);
+        kernelBuilder.AddAzureOpenAIChatCompletion(
+                deploymentName: deploymentName,
+                endpoint: TestConfiguration.AzureOpenAI.Endpoint,
+                apiKey: TestConfiguration.AzureOpenAI.ApiKey);
+        kernelBuilder.Plugins.AddFromType<WidgetFactory>();
+        Kernel kernel = kernelBuilder.Build();
+        kernel.Data["TargetService"] = "AzureOpenAI";
+
+        using var activity = s_activitySource.StartActivity("Main");
+        Console.WriteLine($"Operation Id: {Activity.Current?.TraceId}");
+        if (stream)
+        {
+            await VerifyFunctionCallStreamingAsync(kernel, attempts, functionFilter);
+        }
+        else
+        {
+            await VerifyFunctionCallAsync(kernel, attempts, functionFilter);
+        }
+    }
+
+    #region private
+    private static readonly ActivitySource s_activitySource = new("OpenAI_FunctionCallingTelemetry");
+
+    /// <summary>
+    /// Verify function calling using the provided <see cref="Kernel"/>.
+    /// </summary>
+    private async Task VerifyFunctionCallAsync(Kernel kernel, int attempts, FunctionInvocationFilter functionFilter)
+    {
+        OpenAIPromptExecutionSettings settings = new() { ToolCallBehavior = ToolCallBehavior.AutoInvokeKernelFunctions };
+        IChatCompletionService chatService = kernel.GetRequiredService<IChatCompletionService>();
+        ChatHistory chatHistory = [];
+        int success = 0;
+        Random random = new();
+        Array widgetColors = Enum.GetValues<WidgetColor>();
+        Array widgetTypes = Enum.GetValues<WidgetType>();
+        for (var attempt = 0; attempt < attempts; attempt++)
+        {
+            var widgetColor = (WidgetColor)widgetColors.GetValue(random.Next(widgetColors.Length))!;
+            var widgetType = (WidgetType)widgetTypes.GetValue(random.Next(widgetTypes.Length))!;
+
+            var functionCount = functionFilter.Counter;
+
+            chatHistory.AddUserMessage($"Create a {widgetType} {widgetColor} colored widget for me.");
+            var result = await chatService.GetChatMessageContentsAsync(chatHistory, settings, kernel);
+
+            var resultStr = result.ToString();
+            if (
+                resultStr is not null &&
+                functionFilter.Counter == functionCount + 2 && // Expect the widget factory function to be called
+                resultStr.Contains(widgetColor.ToString(), StringComparison.OrdinalIgnoreCase) && // Expect the color to be in the result
+                resultStr.Contains(widgetType.ToString(), StringComparison.OrdinalIgnoreCase)) // Expect the type to be in the result
+            {
+                success++;
+            }
+        }
+        Assert.Equal(attempts, success);
```

Comment:
nit: I'm not sure we need to write verification logic in concept samples. It would be simpler to read this file without it, just to understand how to enable telemetry for function calling scenario, maybe add some test output etc. In this case, we won't need `Counter` property in function filter. I'm not sure whether `Counter` property is a good example, because here the filter is registered as Singleton, but filter can be also registered as Transient, and this property won't work as expected. I would expect from developers to register filter as Transient in most cases and avoid global properties in real-life scenarios.